dashboard/src/shared/HelmRelease.ts

```diff
@@ -105,35 +127,39 @@ export class HelmRelease {
         acc[releaseName] = cm;
       }
       return acc;
-    }, new Map<string, IHelmReleaseConfigMap>());
+    }, new Map<string, IAppConfigMap>());
 
     // Go through all HelmReleaseConfigMaps and parse as IApp objects
-    const apps = Object.keys(cms).map(key => this.parseRelease(helmReleaseMap[key], cms[key]));
+    const apps = Object.keys(cms).map(key => this.parseRelease(cms[key], helmReleaseMap[key]));
```

Comment:
We need to filter this list of apps by the namespace (if provided) using the namespace in the release data (`app.data.namespace`). Currently we are always showing all Tiller releases despite what namespace is selected. This also has the strange side-effect that a "migrated" app or one installed from the dashboard looks like an un-migrated app (missing HelmRelease CRD) when in a different namespace.

For example, in the default namespace, Redis looks like it hasn't been migrated (icon is missing).
![screen shot 2018-05-30 at 16 48 37](https://user-images.githubusercontent.com/1544861/40753518-66377fa2-6429-11e8-95be-337cf6cf11bb.png)

In the "osba" namespace, it is correct:
![screen shot 2018-05-30 at 16 49 44](https://user-images.githubusercontent.com/1544861/40753533-7b14975c-6429-11e8-9788-45ca3e5dd7a7.png)

With this fix, "redis" and "osba" should not appear at all when in the default namespace, and likewise "default-catalog" and "modest-shrimp" should not appear in the "osba" namespace.

Reply:
good catch!